pybind_interface/GetPybind11.cmake

```diff
@@ -0,0 +1,13 @@
+include(FetchContent)
+
+FetchContent_Declare(
+  pybind11
+  GIT_REPOSITORY https://github.com/pybind/pybind11
+  GIT_TAG v2.2.4
+)
+FetchContent_GetProperties(pybind11)
+find_package(pybind11 CONFIG)
+if((NOT pybind11_FOUND) AND (NOT pybind11_POPULATED)) # check first on system path, then attempt git fetch
```

Comment:
If the local pybind version is older than the requested one, will this recognize that and attempt git-fetching?

Reply:
Fair point. The fix is changing ``find_package(...) -> find_package(pybind11 2.4.4 CONFIG)``, which will do version checking against the installed pybind11 version. For example, here's the section of ``pybind11ConfigVersion.cmake`` checking compatibility of my system ``pybind11 2.6.2`` against a given version (specified as ``PACKAGE_FIND_VERSION`` when doing ``find_package()``).

```cmake
set(PACKAGE_VERSION "2.6.2")

if (PACKAGE_FIND_VERSION_RANGE)
  # Package version must be in the requested version range
  if ((PACKAGE_FIND_VERSION_RANGE_MIN STREQUAL "INCLUDE" AND PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION_MIN)
      OR ((PACKAGE_FIND_VERSION_RANGE_MAX STREQUAL "INCLUDE" AND PACKAGE_VERSION VERSION_GREATER PACKAGE_FIND_VERSION_MAX)
        OR (PACKAGE_FIND_VERSION_RANGE_MAX STREQUAL "EXCLUDE" AND PACKAGE_VERSION VERSION_GREATER_EQUAL PACKAGE_FIND_VERSION_MAX)))
    set(PACKAGE_VERSION_COMPATIBLE FALSE)
  else()
    set(PACKAGE_VERSION_COMPATIBLE TRUE)
  endif()
else()
  if(PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION)
    set(PACKAGE_VERSION_COMPATIBLE FALSE)
  else()
    set(PACKAGE_VERSION_COMPATIBLE TRUE)
    if(PACKAGE_FIND_VERSION STREQUAL PACKAGE_VERSION)
      set(PACKAGE_VERSION_EXACT TRUE)
    endif()
  endif()
endif()
```
I'll push this fix.